packages/sdk/src/utils/Utils.ts

```diff
@@ -31,34 +31,36 @@ export class Utils {
 
     let delimiter = "?"; // initial delimiter
 
-    const cicsPlex = (options && options.cicsPlex) == null ? "" : `${options.cicsPlex}/`;
-    const region = (options && options.regionName) == null ? "" : options.regionName;
+    const cicsPlex = (options && options.cicsPlex) == null ? "" : `${encodeURIComponent(options.cicsPlex)}/`;
+    const region = (options && options.regionName) == null ? "" : encodeURIComponent(options.regionName);
 
     let cmciResource = `/${CicsCmciConstants.CICS_SYSTEM_MANAGEMENT}/${resourceName}/${cicsPlex}${region}`;
 
-    if (options && options.criteria) {
-      cmciResource += `${delimiter}${CicsCmciConstants.CRITERIA}=${this.enforceParentheses(encodeURIComponent(options.criteria))}`;
-      delimiter = "&";
-    }
+    if (options) {
+      if (options.criteria) {
+        cmciResource += `${delimiter}${CicsCmciConstants.CRITERIA}=${this.enforceParentheses(encodeURIComponent(options.criteria))}`;
+        delimiter = "&";
+      }
```

Comment:
Now could be a good time to implement Fernando's efficiency syntax! 💪🏼 

```suggestion
if (options?.criteria) {
    cmciResource += `${delimiter}${CicsCmciConstants.CRITERIA}=${this.enforceParentheses(encodeURIComponent(options.criteria))}`;
    delimiter = "&";
}
// Same for all others
```

Reply:
Will try, bit the reason I changed this was to get around a lint complexity error.